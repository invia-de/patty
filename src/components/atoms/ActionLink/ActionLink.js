import React from 'react';
import PropTypes from 'prop-types';

import cx from '../../../utils/classnames';
import { ArrowDown, ArrowRight } from '../Icon/Icon';

import styles from './actionlink.module.scss';

/**
 * @author [Roman Semko](mailto:roman.semko-extern@invia.de)
 */
export default function ActionLink({ arrow, children, className, ...props }) {
  return (
    <a className={cx(styles.actionlink, className)} {...props}>
      {children}
      {arrow === 'down' ? (
        <ArrowDown className={styles.arrow} />
      ) : (
        <ArrowRight className={styles.arrow} />
      )}
    </a>
  );
}

ActionLink.propTypes = {
  /** additional classNames you want to add */
  className: PropTypes.string,
  /** text of the link */
  children: PropTypes.node.isRequired,
  /** whether to show an arrow at the end of the text */
  arrow: PropTypes.oneOf(['down', 'right']),
  href: PropTypes.string.isRequired
};
