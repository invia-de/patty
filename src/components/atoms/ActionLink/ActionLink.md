```jsx
<ActionLink href="test" arrow="down">
  Go to success page
</ActionLink>
```
