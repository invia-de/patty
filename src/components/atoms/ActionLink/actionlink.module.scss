@import 'theme.scss';

$actionlink-bgcolor-start: $green-medium !default;
$actionlink-bgcolor-end: $green-dim !default;
$actionlink-color: $white !default;

.actionlink,
.actionlink:link,
.actionlink:visited {
  background-image: linear-gradient(
    to bottom,
    $actionlink-bgcolor-start 0%,
    $actionlink-bgcolor-end 100%
  );
  background-repeat: repeat-x;
  padding: 10px;
  font-size: 17px;
  line-height: 24px;
  color: $actionlink-color;
  font-weight: 700;
  text-align: center;
  text-decoration: none;
  margin-top: 0 !important;
}

.arrow {
  margin-left: 10px;
  position: relative;
  top: -1px;
}
