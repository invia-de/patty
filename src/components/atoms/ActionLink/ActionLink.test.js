import React from 'react';
import ActionLink from './ActionLink';
import { render } from '@testing-library/react';

test('ActionLink renders', () => {
  const { container } = render(<ActionLink href="#">Test</ActionLink>);
  expect(container.firstChild).toMatchSnapshot();
});
