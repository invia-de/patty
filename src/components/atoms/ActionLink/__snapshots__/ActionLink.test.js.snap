// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionLink renders 1`] = `
<a
  class="actionlink"
  href="#"
>
  Test
  <svg
    aria-hidden="true"
    class="icon arrow"
    viewBox="0 0 512 512"
  >
    <path
      d="M88 82l176 176L87 432l80 80 259-259L173 0 88 82z"
    />
  </svg>
</a>
`;
